dev/create-dev-env.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2016
+set -euo pipefail
+
+destdir=${1:-nix-bitcoin}
+
+mkdir -p "$destdir/bin"
+cd "$destdir"
+
+if [[ ! -e src ]]; then
+    echo "Cloning fort-nix/nix-bitcoin"
+    git clone https://github.com/fort-nix/nix-bitcoin src
```

Comment:
After cloning, don't you want to `cd` into it?

It seems to me that all this cloning (or not) etc. might just confuse people. Generally I would expect people who run it to already have `gtt clone`d the repo manually. And then having that as pre-condition can be used to double check if the CWD is correct etc.

Reply:
> After cloning, don't you want to cd into it?

No, the setup script simply clones the nix-bitcoin repo to dir `src`, without later changing its contents.

> Generally I would expect people who run it to already have `git cloned` the repo manually

Yes, but if the script moved/renamed the user's existing files, it would be even more confusing.
Users can simply replace the cloned repo with their local clone.

The dev env layout is very opinionated; it's what I prefer for creating auxiliary scripts/notes/resources for a public repo.
But I'm very interested in suggestions on how to improve it.

Reply:
> The dev env layout is very opinionated; it's what I prefer for creating auxiliary scripts/notes/resources for a public repo.
But I'm very interested in suggestions on how to improve it.

Just thinking aloud, I'm not even all that familiar with nix-bitcoin, or generally developing strictly Nix-focused project (as opposed to a Rust project, that just uses Nix for building and tooling).

Generally if the project includes it's own tests and dev-env integration, I would expect `git clone <project-repo> && cd <project> && nix develop (or direnv)` to just set up everything for me (add binaries to the $PATH, create some repo-scoped configs and so on).

Reply:
> Just thinking aloud

Highly appreciated, thanks!

> create some repo-scoped configs

Yeah, like our `scenarios.nix` file. How exactly would you implement this? Where would these files be created? (Note that they must be placed in a separate repo).

Reply:
Note that you can develop nix-bitcoin just fine by simply running `test/run-tests.sh` and editing `tests.nix`.
I'll mention this in the docs.

Reply:
> Yeah, like our scenarios.nix file. How exactly would you implement this? Where would these files be created? (Note that they must be placed in a separate repo).

Oh, I don't really understand how these works, but generally for files like these it's "if they don't exist, copy/ln from a template of some kind and have them in `.gitignore`". [Like here](https://github.com/fedimint/fedimint/blob/4dca27643e673eba166d10ebe4cdbaa4f3fff888/flake.nix#L832)

The "must be placed in a separate repo" seems quite confusing to me, but I wouldn't spend too much time explaining it to me. :D

Reply:
> must be placed in a separate repo

Sorry, that was misleading. What I meant is that files like `scenarios.nix` should be put under version control, but, obviously, in an extra repo. This extra repo could simply be located inside the nix-bitcoin repo, but this has some downsides, as mentioned above.